### PR TITLE
GR: implement treachery and versatile keywords, Com Officer Palik, Elenya the Charming and Curse of Cowardice

### PR DESCRIPTION
**Pull request by strib**

### CODE REVIEW

File: server/game/GameActions/PutIntoPlayAction.js [hunk 1]
```diff
@@ -39,7 +39,10 @@ class PutIntoPlayAction extends CardGameAction {
             return;
         }
 
-        if (card.anyEffect('entersPlayUnderOpponentsControl') && card.owner.opponent) {
+        if (
```
Comment (romanogit) [inline server/game/GameActions/PutIntoPlayAction.js:42]: Take a look at Card.js#setupKeywordAbilities

I believe treachery can be implemented as:

```
this.abilities.keywordPersistentEffects.push(
            this.persistentEffect({
                printedAbility: false,
                condition: () => this.hasKeyword('treachery'),
                match: this,
                effect: AbilityDsl.effects.entersPlayUnderOpponentsControl()
            })
        );
```

File: server/game/costs.js [hunk 2]
```diff
@@ -43,6 +44,8 @@ const Costs = {
                     return true;
                 } else if (context.source.hasHouse(context.player.activeHouse)) {
                     return true;
+                } else if (context.source.hasKeyword('versatile')) {
```
Comment (romanogit) [inline server/game/costs.js:47]: Same here:

```
this.abilities.keywordPersistentEffects.push(
            this.persistentEffect({
                printedAbility: false,
                condition: () => this.hasKeyword('versatile'),
                match: this,
                effect: AbilityDsl.effects.canUse(
                   (card, context, effectContext) => card === effectContext.source
                )
            })
        );
```

Very rough code, but the idea is to use existing effects.

**Review by romanogit — CHANGES_REQUESTED** [review]

File: server/game/GameActions/PutIntoPlayAction.js (hunk 1 above)
Comment (strib) [reply server/game/GameActions/PutIntoPlayAction.js]: Good idea, thanks.  Done.

File: server/game/costs.js (hunk 2 above)
Comment (strib) [reply server/game/costs.js]: Done.

File: server/game/Card.js [hunk 3]
```diff
@@ -49,7 +49,8 @@ class Card extends EffectSource {
             this.printedKeywords[split[0]] = value;
             this.persistentEffect({
                 location: 'any',
-                effect: AbilityDsl.effects.addKeyword({ [split[0]]: value })
+                effect: AbilityDsl.effects.addKeyword({ [split[0]]: value }),
+                name: split[0] + ':printed'
```
Comment (romanogit) [inline server/game/Card.js:53]: was this change on purpose or just to debug?

Comment (strib) [reply server/game/Card.js]: Oops, just to debug. Removed.
